crates/full-node/sov-sequencer/src/preferred/db/mod.rs

```diff
@@ -420,30 +451,41 @@ impl PreferredSequencerDb {
         &self,
     ) -> anyhow::Result<(SequenceNumber, PreferredSequencerCache)> {
         if let Some(backend) = &self.backend {
-            let DbSnapshotData {
-                completed_blobs,
-                in_progress_batch,
-            } = backend.current_data().await?;
-
-            let completed_blobs = VecDeque::from(completed_blobs);
-
-            let sequence_number_of_next_blob = match (completed_blobs.back(), &in_progress_batch) {
-                (Some(blob), None) => blob.sequence_number() + 1,
-                (None, Some(batch)) => batch.sequence_number + 1,
-                (Some(blob), Some(batch)) => {
-                    std::cmp::max(blob.sequence_number(), batch.sequence_number) + 1
-                }
-                (None, None) => 0,
-            };
-
-            Ok((
-                sequence_number_of_next_blob,
-                PreferredSequencerCache::new(
+            match backend.current_data().await {
+                Ok(DbSnapshotData {
                     completed_blobs,
                     in_progress_batch,
-                    self.shutdown_sender.clone(),
-                ),
-            ))
+                }) => {
+                    let completed_blobs = VecDeque::from(completed_blobs);
+
+                    let sequence_number_of_next_blob =
+                        match (completed_blobs.back(), &in_progress_batch) {
+                            (Some(blob), None) => blob.sequence_number() + 1,
+                            (None, Some(batch)) => batch.sequence_number + 1,
+                            (Some(blob), Some(batch)) => {
+                                std::cmp::max(blob.sequence_number(), batch.sequence_number) + 1
+                            }
+                            (None, None) => 0,
+                        };
+
+                    Ok((
+                        sequence_number_of_next_blob,
+                        PreferredSequencerCache::new(
+                            completed_blobs,
+                            in_progress_batch,
+                            self.shutdown_sender.clone(),
+                        ),
+                    ))
+                }
+                Err(DbError::Database(err)) => Err(err),
+                Err(DbError::Replica) => {
+                    tracing::error!(
+                        "initial_data: The primary has become a replica. Shutting down."
+                    );
+                    exit_rollup(&self.shutdown_sender).await;
+                    unreachable!()
+                }
+            }
         } else {
             Ok((
                 0, // TODO this will be revisited when we enable the replica sync task.
```

Comment:
By the way, not related to this PR, but I think we already have the replica sync task enabled right? If the comment is out of date let's just remove it (but if we need more changes then just a heads-up for a future PR)

Reply:
I will continue working on it